app/controllers/loaned_books_controller.rb

```diff
@@ -45,18 +45,15 @@ def assign_new_loan
   def check_quantity_add
     @quantity = total_loan_books
     temp_total_loan = @quantity + User.get_total_loaned_books(current_user)
-    return if temp_total_loan <= Settings.limit_loans.maximum
 
-    quantity_valid @quantity
-    render :new
-  end
-
-  def quantity_valid quantity
-    flash[:danger] = if quantity.negative? ||
-                        quantity.zero?
-                       t "error.quantity.not_valid"
-                      else
-                        t "error.quantity.over"
-                      end
+    if @quantity.negative? || @quantity.zero?
+      flash[:danger] = t "error.quantity.not_valid"
+      redirect_to carts_path
+    elsif temp_total_loan > Settings.limit_loans.maximum
+      flash[:danger] = t "error.quantity.over"
+      redirect_to carts_path
+    else
+      return
```

Comment:
cần 2 line này ko anh bạn, chị đoán là ko nhé
nên xóa đi

Reply:
thực ra chỗ này chị nghĩ em nên tách hẵng 2 method ra

method check_param_quantity_valid và method check_stock_quantity_valid

như vậy chỉ cần if ở 2 method là dc, mà code lại quá ư rõ ràng , clear hơn, method càng to, càng khó handle logic á

Reply:
okie ch